jujugui/static/gui/src/app/components/profile/model-list/model-list.js

```diff
@@ -0,0 +1,256 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+/**
+  Model list React component used to display a list of the users models in
+  their profile.
+*/
+class ProfileModelList extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      loadingModels: false,
+      models: null
+    };
+  }
+
+  componentWillMount() {
+    this._fetchModels(this.props.facadesExist);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const props = this.props;
+    if (props.userInfo.profile !== nextProps.userInfo.profile ||
+        props.facadesExist !== nextProps.facadesExist) {
+      this._fetchModels(nextProps.facadesExist);
+    }
+  }
+
+  /**
+    Makes a request of the controller to fetch the user's availble models.
+    @param {Boolean} facadesExist Whether the controller is connected or not.
+  */
+  _fetchModels(facadesExist) {
+    if (!facadesExist) {
+      console.warn('Controller not connected, skipping fetching models.');
+      return;
+    }
+    // Delay the call until after the state change to prevent race
+    // conditions.
+    this.setState({loadingModels: true}, () => {
+      this.props.listModelsWithInfo(this._fetchModelsCallback.bind(this));
+    });
+  }
+
+  /**
+    Callback for the controller listModels call.
+    @param {String} err The error from the request, or null.
+    @param {Array} modelList The list of models.
+  */
+  _fetchModelsCallback(err, modelList) {
+    this.setState({loadingModels: false}, () => {
+      if (err) {
+        const message = 'Cannot load models';
+        console.error(message, err);
+        this.props.addNotification({
+          title: message,
+          message: `${message}: ${err}`,
+          level: 'error'
+        });
+        return;
+      }
+      // Split the models into ones owned by the active user and ones
+      // shared to the active user.
+      const models = {
+        owned: [],
+        shared: []
+      };
+      modelList.forEach(model => {
+        if (model.owner === `${this.props.userInfo.profile}@external`) {
```

Comment:
Does this work in gijoe and local users?
I guess that rather than just adding `@external` at the end we should be more precise about the domain fo the currently logged in user.